packages/cubejs-client-react/index.d.ts

```diff
@@ -44,6 +44,43 @@ declare module '@cubejs-client/react' {
    */
   export const CubeProvider: React.FC<CubeProviderVariables>;
 
+  type CubeContextVariables = {
+    cubejsApi: CubejsApi;
+  };
+
+  /**
+   * Cube.js context consumer
+   * In case when you need access to cubejsApi directly you can use CubeContext anywhere in your app
+   *
+   * ```js
+   * import React from 'react';
+   * import { CubeContext } from '@cubejs-client/react';
+   *
+   *
+   * export default function DisplayComponent() {
+   *   const { cubejsApi } = React.useContext(CubeContext);
+   *   const [rawResults, setRawResults] = React.useState([]);
+   *   const query = {
+   *     ...
+   *   };
+   *
+   *   React.useEffect(() => {
+   *     const resultSet = await cubejsApi.load(query);
```

Comment:
Almost there :) 

But `useEffect` can't be async on it own. I guess simple `.then` would work here

```js
cubejsApi.load(query).then((resultSet) => ... )
```